src/dune_engine/scheduler.ml

```diff
@@ -671,7 +687,12 @@ let prepare (config : Config.t) ~(handler : Handler.t) =
   Signal_watcher.init events;
   let process_watcher = Process_watcher.init events in
   let t =
-    { status = Building
+    { status =
+        (* Slightly weird initialization happening here: for polling mode we
+           initialize in "Building" state, immediately switch to Standing_by and
+           then back to "Building". It would make more sense to start in
+           "Stand_by" from the start. *)
+        Building
```

Comment:
Why can't we start in `Stand_by`?

Reply:
Ah, the comment is not explaining that. The reason is that then the non-polling code paths will have to look at this variable and switch to Building mode, whereas currently they don't touch it at all. This is probably the right thing to do, but I thought it would require changing more code.

I'll (at least) add to the comment.

Reply:
Added to the comment. Not trying to change the non-polling code paths for now.